### DISCUSSION TITLE
If the list doesn't fit the screen it sometimes ignores the order configuration.

### DISCUSSION THREAD

**Discussion by GitUser0080** [marlonrichert/zsh-autocomplete#190] [Q&A] [answered]
* `zsh-autocomplete` version: February 5 commit
* Zsh version: 5.8
* Framework: none
* Plugin manager: none

```zsh
$ cd $(mktemp -d)  # Create a temp dir and enter it.
$ ZDOTDIR=$PWD HOME=$PWD zsh -f  # Start a subshell in it without config files.
% source path/to/zsh-autocomplete.plugin.zsh  # Source the plugin.
% cd ctftrash # Go to the directory with files
% clear
% ls
% cat s # To this point everything works correctly
% # Now do long listing
% ls -l
% cat s # only the directories appear
% # if you remove 's' and type it again eventually all files will appear
```
I don't know if it's an intended behavior but since it happens to me I decided to report it. When a list doesn't fit the screen it displays only "local directories" regardless of configuration. One thing to note is that I specified in my .zshrc to display the "file" before the "local directory". This however seems to have no effect in this case. 
Here's a video:
<a href="https://asciinema.org/a/QvADKx2ez0vi8lB8iZdqUoE37" target="_blank"><img src="https://asciinema.org/a/QvADKx2ez0vi8lB8iZdqUoE37.svg" /></a>
Of course, I realize that I can just click "Ctrl-C" to display the whole list, but that requires me an extra click and is not really an intended solution. In a situation like this, where the whole list doesn't fit I would rather expect to see the "files" in autocomplete list as this what I configured. The other thing is that available space technically expands before it makes a suggestion for "cat s". It doesn't recognize that though which is why on the above video after I deleted the "s" character and typed it again I still got only 3 files on the list even though it should show 11. 
I don't know if it's intended but for me, it looks like an issue.
You can just recreate blank files with this script - <a href="https://pastebin.com/4gnCL60a">Pastebin</a>
My .zshrc - <a href="https://pastebin.com/CE5NYEPp">Pastebin</a>

**Comment by marlonrichert** [comment, marked as answer]
This is because while `group-order` controls the order in which completions are displayed, it does not control the order in which they are _generated._ For file completions, this is controlled by the [`file-patterns` setting](http://zsh.sourceforge.net/Doc/Release/Completion-System.html#index-file_002dpatterns_002c-completion-style). To get what you want, you need to use the following:
```zsh
zstyle ':completion:*:' group-order \
  options executables \
  globbed-files directories all-files \
  suffix-aliases aliases functions builtins reserved-words
zstyle ':completion:*' file-patterns \
  '%p(-.):globbed-files *(-/):directories:directory' '*:all-files'
```
